src/directwrite/mod.rs

```diff
@@ -238,12 +230,24 @@ impl crate::Rasterize for DirectWriteRasterizer {
     fn get_glyph(&mut self, glyph: GlyphKey) -> Result<RasterizedGlyph, Error> {
         let loaded_font = self.get_loaded_font(glyph.font_key)?;
 
-        match self.rasterize_glyph(&loaded_font.face, glyph.size, glyph.c) {
-            Err(err @ Error::MissingGlyph(_)) => {
-                let fallback_font = self.get_fallback_font(&loaded_font, glyph.c).ok_or(err)?;
-                self.rasterize_glyph(&fallback_font.create_font_face(), glyph.size, glyph.c)
-            },
-            result => result,
+        let loaded_fallback_font;
+        let mut font = loaded_font;
+        let mut glyph_index = self.get_glyph_index(&loaded_font.face, glyph.character);
+        if glyph_index == MISSING_GLYPH_INDEX {
+            if let Some(fallback_font) = self.get_fallback_font(&loaded_font, glyph.character) {
+                loaded_fallback_font = Font::from(fallback_font);
+                glyph_index = self.get_glyph_index(&loaded_fallback_font.face, glyph.character);
+                font = &loaded_fallback_font;
+            }
+        }
```

Comment:
> You can't return a reference to a stack local object from a method, like get_fallback_font, so this fallback should be stored somewhere, and I don't see where you'll store it with your approach.

Have you looked at my suggested code? It does exactly that.

Reply:
I looked, and your example only works because of the fact that you create and return right in the end of method (I'm not even sure why rust allows that...). In our case we can't do that, I've tried and it fails with the correct reason for that
```
error[E0515]: cannot return value referencing temporary value
   --> src/directwrite/mod.rs:122:9
    |
122 |            fallback.map_characters(
    |   _________^
    |  |_________|
    | ||
123 | ||             &text_analysis_source,
124 | ||             0,
125 | ||             length,
...   ||
130 | ||             loaded_font.stretch,
131 | ||         ).mapped_font.map(Font::from).as_ref()
    | ||_____________________________________-________^ returns a value referencing data owned by the current function
    | |______________________________________|
    |                                        temporary value created here

```

Your example will also fail if you do `let m = Font; Some(&m)` for the exact reason.,

Reply:
A more complete example

```rust
fn get_fallback_font<'a>(&'a self, loaded_font: &'a Font, character: char) -> Option<&'a Font> {
    if character == '\0' {
        return Some(loaded_font);
    }
    let fallback = self.fallback_sequence.as_ref()?;

    let mut buffer = [0u16; 2];
    character.encode_utf16(&mut buffer);

    let length = character.len_utf16() as u32;
    let utf16_codepoints = &buffer[..length as usize];

    let locale = get_current_locale();

    let text_analysis_source_data = TextAnalysisSourceData { locale: &locale, length };
    let text_analysis_source = TextAnalysisSource::from_text(
        Box::new(text_analysis_source_data),
        Cow::Borrowed(utf16_codepoints),
    );

    let fallback_result = fallback.map_characters(
        &text_analysis_source,
        0,
        length,
        &self.available_fonts,
        Some(&loaded_font.family_name),
        loaded_font.weight,
        loaded_font.style,
        loaded_font.stretch,
    ).mapped_font?;

    Some(&Font::from(fallback_result))
}
```

No, it doesn't work, because you refer to local data.